include/DREAM/Equations/Fluid/IonNeutralAdvectionDiffusionTerm.tcc

```diff
@@ -0,0 +1,67 @@
+#include "DREAM/Constants.hpp"
+#include "DREAM/Equations/Fluid/IonNeutralAdvectionDiffusionTerm.hpp"
+#include "DREAM/IonHandler.hpp"
+//#include "DREAM/NotImplementedException.hpp"
+#include "FVM/Grid/Grid.hpp"
+
+
+using namespace DREAM;
+
+template<class T>
+IonNeutralAdvectionDiffusionTerm<T>::IonNeutralAdvectionDiffusionTerm(FVM::Grid *grid, IonHandler *ihdl, const len_t iIon, bool allocCoefficients
+	) : IonEquationTerm<T>(grid, ihdl, iIon, allocCoefficients){}
+
+template<class T>
+IonNeutralAdvectionDiffusionTerm<T>::~IonNeutralAdvectionDiffusionTerm(){}
+
+template<class T>
+bool IonNeutralAdvectionDiffusionTerm<T>::SetCSJacobianBlock(
+    const len_t uqtyId, const len_t derivId, FVM::Matrix *jac, const real_t *f,
+    const len_t /*iIon*/, const len_t Z0, const len_t rOffset
+){
+    bool contributes = false;
+    
+    if(!this->HasJacobianContribution(derivId) || Z0>0)
```

Comment:
Switch the order of these checks. If statements are always evaluated left -> right, and comparing an integer to a constant is always way faster than calling ``HasJacobianContribution()``.